src/llms/openai.ts

```diff
@@ -273,6 +290,21 @@ export function accumulateContent(): UnaryFunction<Observable<ChatCompletionsRes
   );
 }
 
+/**
+ * An rxjs operator that handles errors in a stream of chat completion responses.
+ *
+ * When the first error is encountered, the callback will be called with the error message
+ * and the stream will be stopped.
+ */
+export function handleError(callback: (s: string) => void): UnaryFunction<Observable<ChatCompletionsResponse<ChatCompletionsChunk> | ChatCompletionsErrorResponse>, Observable<ChatCompletionsResponse<ChatCompletionsChunk>>> {
```

Comment:
What about throwing an error on the pipe with the error message? So the people consuming the stream can use the standard `catchError` rxjs function on the pipe?

This will also allow using it as 
```js
const stream = llms.openai
      .streamChatCompletions({
        model,
        temperature,
        messages,
      })
      .pipe(
        // Accumulate the stream content into a stream of strings, where each
        // element contains the accumulated message so far.
        llms.openai.accumulateContent()
        // The stream is just a regular Observable, so we can use standard rxjs
        // functionality to update state, e.g. recording when the stream
        // has completed.
        // The operator decision tree on the rxjs website is a useful resource:
        // https://rxjs.dev/operator-decision-tree.
      );
      stream.subscribe({
        next: setReply,
        // THIS WON'T BE CALLED, because the error comes as a message
        error: (e)=> console.log(e),
        complete: () => {
          setIsGenerating(false);
          setMessages([]);
        },
      })
```

I think, if we hide the error in the error message type, would confuse people why `catchError` or

Reply:
This is working well for me in the example plugin 👍 thanks for the feedback!